lib/src/cpp/utils/XMLManager.cpp

```diff
@@ -177,9 +174,13 @@ void XMLManager::transform_standalone_to_rooted_structure()
 
 void XMLManager::validate_and_save_document()
 {
+    // Set the write required as false by default
+    write_required = false;
```

Comment:
This flag has already been set to false in the `initialize` method. Publicly, the user can only call `initialize` and `terminate`, so I do not think it requires to be set to false again here. That or just set the flag to false in this function.